File: runtime/typescript/src/I16.ts
import { Codec } from "./Codec.js";
import { FormatReader, FormatWriter } from "./FormatIO.js";

export type I16 = number;

export const codec: Codec<I16> = {
    async read(reader: FormatReader): Promise<I16> {
        return (await reader.readU16() << 16) >> 16;
    },

    write(writer: FormatWriter, value: I16): Promise<void> {
        return writer.writeU8(value & 0xFFFF);
    },
};

File: runtime/java/src/main/java/dev/argon/verilization/runtime/I16.java
package dev.argon.verilization.runtime;

import java.io.IOException;

public final class I16 {
    private I16() {}

    public static short fromInteger(int i) {
        return (short)i;
    }

    public static final Codec<Short> codec = new Codec<Short>() {
        @Override
        public Short read(FormatReader reader) throws IOException {
            return reader.readShort();
        }

        @Override
        public void write(FormatWriter writer, Short value) throws IOException {
            writer.writeShort(value);
        }
    };
}
